app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeScreen.kt

```diff
@@ -68,6 +70,16 @@ fun HomeScreen(
     val showTrendingCoinsLoading: LoadingState by viewModel.output.showTrendingCoinsLoading.collectAsState()
     val myCoins: List<CoinItemUiModel> by viewModel.output.myCoins.collectAsState()
     val trendingCoins: List<CoinItemUiModel> by viewModel.output.trendingCoins.collectAsState()
+    val myCoinsError: Throwable? by viewModel.output.myCoinsError.collectAsState()
+    val trendingCoinsError: Throwable? by viewModel.output.trendingCoinsError.collectAsState()
+
+    myCoinsError?.let { error ->
+        Toast.makeText(context, error.userReadableMessage(context), Toast.LENGTH_SHORT).show()
+    }
+
+    trendingCoinsError?.let { error ->
+        Toast.makeText(context, error.userReadableMessage(context), Toast.LENGTH_SHORT).show()
+    }
```

Comment:
Why do we need to separate the error?

Reply:
@luongvo Usually, we will separate the error stream for the case that can happen at the same time. For example:

When calling the `getMyCoins` and `getTrendingCoins`, an error at the same time, and we use the same stream`(_error)` to emit the `error`. For currently, It's still good because we just show the `Toast`, but what about if we need to render some UI state`(empty state)` in each section of `MyCoins` and `Trending`, That's why I think we should recommend separating error from the normal one in `BaseViewModel`, It could be easy for handling the UI 

Another reason is the base one in `BaseViewModel` is a `SharedFlow`, but if we consider the behavior of calling the `MyCoins/TrendingCoins` list, It seems like it's a `StateFlow` because it can refresh, that means it can be error any time once refresh 🤔 

And one more thing regarding `refreshing` behavior, Should we handle `viewModel.output.error` inside `LaunchedEffect` scope?  If I'm not wrong, The `LaunchedEffect` should handle the state only one time when `entering/leaving` from the screen 🤔 

That's all my concerns to make me decide to separate it and use `StateFlow` instead `SharedFlow`. Please correct me if I missed something? cc @hoangnguyen92dn

Reply:
@luongvo If I understand right Pooh's idea, she wants to handle errors in each section separately rather than combine them into a single error stream then; it will be more complex when we need to handle the specific error in case one of these coin lists is failed, and the other is successful.

Reply:
Yes @luongvo like @hoangnguyen92dn said, I want to handle errors in each section separately rather than combine them into a single error stream 🙏